src/core/mount.c

```diff
@@ -1646,7 +1646,7 @@ static int mount_setup_existing_unit(
                 /* If things changed, then make sure that all deps are regenerated. Let's
                  * first remove all automatic deps, and then add in the new ones. */
 
-                unit_remove_dependencies(u, UNIT_DEPENDENCY_MOUNTINFO_IMPLICIT);
+                unit_remove_dependencies(u, UNIT_DEPENDENCY_MASK_MOUNTINFO);
```

Comment:
So basically you now remove the deps involved with `DefaultDependencies=`. Given that these deps depend on the present of a unit file, I'm not sure to follow...

Could you explain ? (and add the rational in the commit message ?)

Reply:
Also I'm not sure that hiding the list of bits(only 2 in this case) behind a new "mask" macro helps in term of readability.